### DISCUSSION TITLE
CBZ/CBR thumbnail support

### DISCUSSION THREAD

**Discussion by rudexi** [spacedriveapp/spacedrive#2456] [Feature Request]
Hi team,

I'd like to suggest the support of thumbnails for CBZ/CBR formats (for comic/manga).
For format itself is a simple zip/rar file, containing images, and renamed into .cbz/.cbr, respectively.
Usually, most explorers supporting it display the first image found in the archive (by alphabetic order).

So far on Windows explorer, I was using this project to display CBZ thumbnails
https://github.com/T800G/CBXShell

I have some experience on Rust and Typescript, so I'm willing to work on it. I'd just need some hints on how to add a new thumbnail format.

**Comment by utkubakir** [comment]
Great idea! 

@fogodev would be your guy for the hints. I also recommend joining our discord server for easier communication :)

**Comment by fogodev** [comment]
Here, in this function, we dispatch an appropriate handler for the desired extension. So I think that to support it we only need to create a new handler here, and then pass the new extensions to be allowed in the thumbnailer.

https://github.com/spacedriveapp/spacedrive/blob/44478207e72495b3777e294660d78939711b544f/crates/images/src/handler.rs#L29-L67

**Reply by fogodev** [reply]
Feel free to reach me out on discord for any further explanations

**Reply by rudexi** [reply]
Thank you, will take a look at the code and come back to you on discord when I need further details.
